third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -710,6 +711,67 @@ class ConvertSCFForOp : public PointerCanonicalizationPattern<scf::ForOp> {
   }
 };
 
+/// Rewrite warp parition args.
+class ConvertWarpSpecializeOp
+    : public PointerCanonicalizationPattern<ttg::WarpSpecializeOp> {
+  using PointerCanonicalizationPattern::PointerCanonicalizationPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite_(ttg::WarpSpecializeOp wsOp, OneToNOpAdaptor adaptor,
```

Comment:
I am wondering if convertSimpleBlockSignature be used to simplify the code in this function?